Source/WebCore/svg/SVGGradientElement.cpp

```diff
@@ -80,6 +82,15 @@ void SVGGradientElement::svgAttributeChanged(const QualifiedName& attrName)
 {
     if (PropertyRegistry::isKnownAttribute(attrName) || SVGURIReference::isKnownAttribute(attrName)) {
         InstanceInvalidationGuard guard(*this);
+
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+        if (document().settings().layerBasedSVGEngineEnabled()) {
+            if (auto* gradientRenderer = dynamicDowncast<RenderSVGResourceGradient>(renderer()))
+                gradientRenderer->invalidateGradient();
+            return;
+        }
+#endif
```

Comment:
This section seems repeated, should we have a helper method on SVGGradientElement?

Reply:
Good catch, will use it from SVGLinear/SVGRadialGradientElement as well.